apis/v1alpha1/instrumentation_types.go

```diff
@@ -113,6 +113,10 @@ type Java struct {
 	// If the former var had been defined, then the other vars would be ignored.
 	// +optional
 	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Resources describes the compute resource requirements.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
Is there a reason to put these on the individual language structs rather than `InstrumentationSpec` Is the idea that different languages may have different resource requirements?

Reply:
So what you want to see is another config on the InstrumentationSpec level that gets overwritten by the language specific one?

Reply:
That would keep things consistent because we allow that pattern with environment variables.  Being able to set it at the root level would let some users set the resources in one spot for all languages, reducing their config.  

I recognize that there would be a conflict at the root level tho, and that it doesn't save that much config, so I am ok if we stick with language-only.  It could also always be added later in a future PR.

Reply:
+1 to what tyler is recommending here, would also be fine if we put that in a future PR :)

Reply:
> ...reducing their config.

I hope that the default values will ensure that in most cases no further settings are needed. ^^